### ISSUE TITLE
Pretrained weights

### ISSUE THREAD

**Issue by hichemfelouat** [neeek2303/MegaPortraits#1]
Hi,
When do you intend to publish the pretrained model for testing?

**Comment by 2blackbar** [comment]
why whould they , for free. they turn it into some filter on a phone

**Comment by Kevyunsun** [comment]
Any idea when this may become available? I would love to pay for the ability to use this. I have so many ideas!

**Comment by FrancescoSaverioZuppichini** [comment]
What about the code? It would be great to actually evaluate if the results in the paper are true

**Comment by LeoLovechest** [comment]
> Any idea when this may become available? I would love to pay for the ability to use this. I have so many ideas!

It is my belief this method of 2D image animation is a poorer quality to what is currently available & they have just trained this model with a higher resolution.

I have been working with alternatives & would love to hear your ideas as I am looking for more use cases.
Feel free to message me on Discord **Leo Lovechest#7410**

**Comment by FrancescoSaverioZuppichini** [comment]
so, guys, it looks like we should trust them they are reporting the real result in the paper - science in 2023 :)

**Comment by khakhulin** [comment]
Hi @FrancescoSaverioZuppichini ! I understand that pretrained weights can decrease the barrier to compare with those results. However, in such cases, you can always contact the authors directly if the goal is to compare with the method. For example, in this paper, the 2 main baselines did not publish models, but their results are present in this paper. To write that you doubt the reality of the results, just because the model is not presented here for personal usage, is not persuasive, since this does not create huge obstacles for academic validation of the result

**Comment by FrancescoSaverioZuppichini** [comment]
Hi @khakhulin thanks a lot for your reply, I am not an academic myself but I believe in open science and if we can't use it, then well, it's just printed words :) That's a pity, I wanted to do a harry potter animated paintings repo :(

**Comment by FrancescoSaverioZuppichini** [comment]
Moreover, if you don't want to share your with the community your work. What's the point of having a repo on GitHub? Does it serve another purpose that is not like grabbing? Thanks a lot again, have a nice one :)

**Comment by LeoLovechest** [comment]
> Moreover, if you don't want to share your with the community your work. What's the point of having a repo on GitHub? Does it serve another purpose that is not like grabbing? Thanks a lot again, have a nice one :)

See my previous message to another user, as I am doing similar stuff to you it's worth a chat so I can share my current results using other tools & explain my process.

**Comment by FrancescoSaverioZuppichini** [comment]
@LeoLovechest Sure, let's join forces. Feel free to DM via email: francesco.zuppichini@gmail.com 

Again, @khakhulin science is collaboration, not just sharing SOTA results on a latex table IMHO :angel:

**Comment by neeek2303** [comment]
Hi @FrancescoSaverioZuppichini, you can believe in whatever you want, but, unfortunately, reality is different. Let me explain you briefly how this works, probably is hard to understand if you are not into academia. 

A lot of companies do research and don't share any information about it, not even a paper. They spend a lot of money on this and want to keep technologies private to earn money (this is how capitalism works). Some of them release papers for some works, sometimes they release codes with papers. We, authors, from our side, provide everything that we can, under current company policy, to share results with community. I know people who reproducing this paper, yes it is harder than just take weights and use it for free, but it is not a rocket science. This type of research are not suppose to be open source. I would glad if all companies share all their technologies on which they spend tones of money, but as I said reality is different. Again, if you have any ideas how to change this system, you can text bigtech companies with your  suggestions directly.

However, research that is done in real academia (universities), in vast majority of cases are open sourced, because most universities have such a policy. I've just entered a PhD program and plan to do open source projects in the future. Keep tuned.

Btw, this is a repo for our website https://samsunglabs.github.io/MegaPortraits/. To create this you need a github repo.
Have a nice day🌞🌞🌞

**Comment by FrancescoSaverioZuppichini** [comment]
Hey @neeek2303, thanks for the highlights - not liking your tone but why not distribute the software with a non-commercial license? I don't want to assume you don't really care about openness but I really don't understand why making a website etc if you are not open to sharing your work. 

I understand most people in academia only want to show not to share, but still ... what a waste! 

P.S. Microsoft come one, add the :hankey: emoji as a reaction!

**Comment by neeek2303** [comment]
@FrancescoSaverioZuppichini I also don't like your words that our work is just latex table and printed paper, you have got a tone, that you started the conversation with. As I said you can ask this question to company directly, I don't have answers for you and I can't represent the company here.

**Comment by FrancescoSaverioZuppichini** [comment]
@neeek2303 thanks for the reply! Apologies if you felt attacked somehow, but as I said, well-written paper, nice numbers and cool GIFs, but that's it - or at least IMHO

Side question: in your opinion, what's the incentive in making a repo and the website (two tools used to **share**) if nothing is really there and/or available to the public?

**Comment by ttorpp** [comment]
I think [HuggingFace](https://huggingface.co/) can become a new GitHub for sharing a trained model and data.
And [Gradio](https://gradio.app/) would be a good place for a simple free demo. 

I fully understand why authors didn't share the model and code. 

But, as long as the paper is published, hiding the model and code is ineffectice strategy in a capitalism viewpoint. That's becauce any serious competitors (companies) can easily reproduce the result. They're not an individual. They have enough money, employees, and resources to do so.

But for individual researchers, it's an extra burden if the model and code is not shared.

In conclusion, only publishing paper without the model and code does not have any negative effect to the competitors, but to individual researchers with goodwill.

If you share them, then individual researchers will feel gratitude, which is also good for company's brand image (Many companies do good social contribution for their brand image).

Sincerely,

**Comment by FrancescoSaverioZuppichini** [comment]
For what is worth, without any weights to check the results, the paper has the same scientific validity of the book of genesis - you need to be a believer

**Comment by jjangga0214** [comment]
> If you share them, then individual researchers will feel gratitude, which is also good for company's brand image (Many companies do good social contribution for their brand image).

I agree especially from a developer's point of view. For example, Google open-sourced Chromium in 2008. This made a developer-friendly, free, and cool image for Google, while MS with IE built an uncool, stubborn, and behind-the-times image.

Developers always have shown somewhat more innovative movement compared to other social fields. They've shared code. They've contributed to open source. They spent a large amount of time and energy. Sometimes they were purely useful, some other times they were literally destructive innovations. Anyway we all eventually realized this openness showed a great aspect of humanity, leading to the progress of history. 

Samsung and MegaPortrait are also **"standing on the shoulders of giants (open source)"**. People would be happy if you **give back** to the community. 

I understand the culture of ML researchers is not 100% the same as that of developers. Some of researchers have a certain amount of academic characteristics and backgrounds. There has been a rather strict, credentialed, capitalist atmosphere in their culture. Patents, lawsuits, defense of the rights, or the so-called Honor-system would be such things.

I am worried that the greatness of the open-source culture is contaminated by such a mindset these days, which is frequently shown in ML fields.

I am not saying sharing every model would be great. For instance, ChatGPT requires hundreds of millions of dollars to train and experiment, and MS wants to make money from the service. I understand they just publish a conceptual paper without any real detail. (_Even though Meta open-sourced LLAMA!_ (Meta did it, by the way. Why not Samsung?))

**But as long as I understand correctly, MegaPortraits is far from such cases. MegaPortraits is a much smaller model.** Compared to other popular researches, the one significant unique point of MegaPortraits is a different idea of utilizing latent space. **People are just interested whether this new theoretical idea really works well.** Nothing more, nothing less.  

> In conclusion, only publishing paper without the model and code does not have any negative effect to the competitors, but to individual researchers with goodwill.

I can't but agree with this statement. As MegaPortraits is simple, competitors would easily train it.

**I believe in the faith that humanity progresses the more than the side effect eventually when knowledge and technology are shared with the public.**

Thank you for the research, and I hope you let people verify if your research is really a breakthrough or not.

Maybe you can have (request) an internal meeting (with a boss) to change the traditional company rules. Hope you can persuade.

**Comment by johndpope** [comment]
fyi - one of the lead researchers that worked on this paper from samsung ai - egor zakharov https://egorzakharov.github.io/ - has new / updated code dropping soon
https://p0lyfish.github.io/voodoo3d/ - looks promising - 

@egorzakharov 
https://github.com/egorzakharov
